### PR TITLE
Update debian and centos packers to use i386 architecture

### PR DESCRIPTION
**Pull request by ripcurld0**
A follow up to #6156

### CODE REVIEW

**Comment by ripcurld0** [comment]
@andrewkroh 
I tried to figure it out how to test that as a whole but it took me too much so I ran small tests to make sure it works.
Nevertheless, I want to be 100% sure it does what it does. Do you have any idea?

File: auditbeat/Makefile [hunk 1]
```diff
@@ -33,10 +33,15 @@ before-build:
...
 	     ${ES_BEATS}/auditbeat/_meta/common.p2.yml \
 	     ${ES_BEATS}/libbeat/_meta/config.yml > \
 	     ${PREFIX}/${BEAT_NAME}-linux.yml
+	@cat ${ES_BEATS}/auditbeat/_meta/common.p1.yml \
+	     <(go run scripts/generate_config.go -os linux -arch i386 -concat) \
+	     ${ES_BEATS}/auditbeat/_meta/common.p2.yml \
+	     ${ES_BEATS}/libbeat/_meta/config.yml > \
+	     ${PREFIX}/${BEAT_NAME}-i386-linux.yml
 	@cat ${ES_BEATS}/auditbeat/_meta/common.reference.yml \
 	     <(go run scripts/generate_config.go -os linux -concat -ref) \
```
Comment (andrewkroh) [inline auditbeat/Makefile]: We will also need to create an architecture specific "reference" config file for Linux too. (e.g. `${BEAT_NAME}-linux-i386.reference.yml`).

File: dev-tools/packer/platforms/centos/run.sh.j2 [hunk 2]
```diff
@@ -35,7 +39,7 @@ FPM_ARGS=(
         homedir/=/usr/share/{{.beat_name}}
         beatname-${RUNID}.sh=/usr/bin/{{.beat_name}}
         {{.beat_name}}-linux-{{.arch}}=/usr/share/{{.beat_name}}/bin/{{.beat_name}}
-        {{.beat_name}}-linux.yml=/etc/{{.beat_name}}/{{.beat_name}}.yml
+        ${BEATS_YML_NAME}.yml=/etc/{{.beat_name}}/{{.beat_name}}.yml
         {{.beat_name}}-linux.reference.yml=/etc/{{.beat_name}}/{{.beat_name}}.reference.yml
```
Comment (andrewkroh) [inline dev-tools/packer/platforms/centos/run.sh.j2]: Can you reference the `BEATS_YML_NAME` var for the reference yml too.

**Comment by andrewkroh** [comment]
FYI Once this is merged to master there is a CI job that will build the new packages and publish them to S3.

- Jenkins Job: https://beats-ci.elastic.co/job/elastic+beats+master+package/
- Published artifacts: https://s3-us-west-2.amazonaws.com/beats-package-snapshots/index.html

After publishing there is another job that will attempt to install them on various OSes and smoke test.

- beats-tester job: https://beats-ci.elastic.co/job/elastic+beats-tester+master/
- beats-tester project: https://github.com/elastic/beats-tester

**Comment by ripcurld0** [comment]
@andrewkroh thanks a lot for the review it helped me a lot testing the changes.
Sorry it took me a while, though.

**Comment by ripcurld0** [comment]
Travis failed only in two jobs:

1) https://travis-ci.org/elastic/beats/jobs/337341238
> --2018-02-04 23:06:46--  https://snapshots.elastic.co/downloads/elasticsearch/elasticsearch-7.0.0-alpha1-SNAPSHOT.tar.gz.sha512
> Resolving snapshots.elastic.co (snapshots.elastic.co)... 184.73.245.233, 184.73.156.41, 184.72.242.47, ...
> Connecting to snapshots.elastic.co (snapshots.elastic.co)|184.73.245.233|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
> 2018-02-04 23:06:46 ERROR 404: Not Found.
> /bin/sh: line 0: test: ==: unary operator expected

2) https://travis-ci.org/elastic/beats/jobs/337341251
> Step 12/20 : RUN curl -Ls ${DOWNLOAD_URL}/kibana/kibana-${ELASTIC_VERSION}-linux-x86_64.tar.gz?c=${CACHE_BUST} | tar --strip-components=1 -zxf - &&     ln -s /usr/share/kibana /opt/kibana
>  ---> Running in 2f8b2b7574af
> gzip: stdin: not in gzip format
> tar: Child returned status 1
> tar: Error is not recoverable: exiting now

Those two look like an infra problem and not related to code.

**Comment by tsg** [comment]
Those can be safely ignored for the purposes of this PR. I'm going to give this a quite local test.

**Comment by tsg** [comment]
I tested this one manually for deb, rpm, and tar.gz and looks good to me.

One thing I noticed is that we customise for 32-bit the default (short) configurations, but not the reference configurations (`auditbeat.reference.yml`).  I think we should merge this PR as is, and consider if we want to follow up with another PR to also update the reference files.

**Comment by ripcurld0** [comment]
Thank you @tsg for reviewing and merging 🥇 
Yep, I'm going to work on that too.

**Pull request merged**